src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -192,6 +188,12 @@ internal static class ParameterHelpers
                     diagnostics.Add(ErrorCode.ERR_IllegalRefParam, refnessKeyword.GetLocation());
                 }
 
+                if (scope == DeclarationScope.Unscoped &&
+                    IsRefScopedByDefault(refKind, parameterType))
+                {
+                    scope = DeclarationScope.RefScoped;
```

Comment:
I thought that in one of Julien's PRs we moved to make this calculation occur for out parameters in the `EffectiveScope` accessor instead. Was that the case? Now it seems like it is factored into the `DeclaredScope` again. I just want to make sure we aren't applying the logic redundantly and we feel like we are putting it in the right place in this PR.

Reply:
That PR, [https://github.com/dotnet/roslyn/pull/63134](https://github.com/dotnet/roslyn/pull/63134), has not be merged yet.